src/cascadia/TerminalSettingsModel/EnumMappings.cpp

```diff
@@ -59,4 +58,24 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         }();
         return enumMap;
     }
+
+    // BellStyle is special because we deprecated 'visual' for the settings UI but
+    // want to allow it for legacy users
+    winrt::Windows::Foundation::Collections::IMap<winrt::hstring, Model::BellStyle> EnumMappings::BellStyle()
+    {
+        static IMap<winrt::hstring, Model::BellStyle> enumMap = []() {
+            auto map = single_threaded_map<winrt::hstring, Model::BellStyle>();
+            for (auto [enumStr, enumVal] : JsonUtils::ConversionTrait<Model::BellStyle>::mappings)
+            {
+                const auto enumHStr = winrt::to_hstring(enumStr);
+                // skip visual
+                if (enumHStr != L"visual")

```

Comment:
if you want to remove manual code from this file, you could define a copy of `DEFINE_ENUM_MAP`, name it `DEFINE_FLAG_MAP`, and add the same `WI_IsSingleFlagSet` check to it. I think this would also automatically handle the `none` value (which we do not want to emit an enum entry for).

Reply:
So actually, turns out that while we're sticking to the manual checkbox creation/selection we don't actually need this at all. Removed it for now!